src/components/MainContext/MainContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+import { Errors } from '../../types/Errors';
+import { ContextProps } from '../../types/ContextProps';
+
+export const MainContext = React.createContext<ContextProps>({
+  users: [],
+  posts: [],
+  setPosts: () => { },
+  choosedPost: null,
+  setChoosedPost: () => { },
+  comments: [],
+  setComments: () => { },
+  error: Errors.NONE,
+  setError: () => { },
+  isLoading: false,
+  setIsLoading: () => { },
+  isActiveComForm: false,
+  setIsActiveComForm: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const MainProvider: React.FC<Props> = ({ children }) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [choosedPost, setChoosedPost] = useState<Post | null>(null);
```

Comment:
it's a correct third from of the verb `choose`
```suggestion
  const [chosenPost, setChosenPost] = useState<Post | null>(null);
```